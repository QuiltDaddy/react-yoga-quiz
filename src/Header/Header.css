header {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  height: 90px;
  max-width: 750px;
  margin-left: auto;
  margin-right: auto;
}
