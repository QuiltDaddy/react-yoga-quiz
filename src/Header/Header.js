import React, { Component } from 'react';
import './Header.css';

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      width: window.innerWidth
    }

    this.widthChange = this.widthChange.bind(this);
  }

  componentDidMount() {
    window.addEventListener('resize', this.widthChange);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.widthChange);
  }

  widthChange() {
    this.setState({
      width: window.innerWidth
    });
  }

  render() {
    const {question, score, topScore} = this.props;

    return (
      <header role='banner'>
        {(this.state.width > 430) ? (
          <>
            <h2>Question: {question}</h2>
            <h1>Yoga Quiz</h1>
            <h2>Score: {score}/{topScore}</h2>
          </>
        ) : (
          <>
            <h1>Yoga Quiz</h1>
            <div>
              <h2>Question: {question}</h2>
              <h2>Score: {score}/{topScore}</h2>
            </div>
          </>
        )}
      </header>
    );
  }
}

export default Header;
