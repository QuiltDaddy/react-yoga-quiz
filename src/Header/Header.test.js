import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from './Header';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
const tree = renderer
  .create(
    <Header
      question={1}
      score={0}
      topScore={1} />)
  .toJSON();
expect(tree).toMatchSnapshot();  
});
