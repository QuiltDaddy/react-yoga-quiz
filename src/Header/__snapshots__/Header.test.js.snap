// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<header
  role="banner"
>
  <h2>
    Question: 
    1
  </h2>
  <h1>
    Yoga Quiz
  </h1>
  <h2>
    Score: 
    0
    /
    1
  </h2>
</header>
`;
